### PR TITLE
Add log search

### PR DESCRIPTION
**Pull request by teodor-ritense**
[[FE] Searching technical logs via the UI](https://ritense.tpondemand.com/entity/118281-fe-searching-technical-logs-via-the)

*PLEASE CHECK TRANSLATIONS*

### CODE REVIEW

File: projects/valtimo/config/assets/core/nl.json [hunk 1]
```diff
@@ -1856,6 +1856,16 @@
...
       "level": "Logniveau"
     },
+    "search": {
+      "addProperty": "Eigenschap toevoegen",
+      "afterTimestamp": "Van datum",
+      "beforeTimestamp": "Tot datum",
+      "properties": "Eigenschappen"
+    },
+    "noResults": {
+      "description": "Er zijn geen logboeken die overeenkomen met uw zoekcriteria",
```
Comment (mbritense) [inline projects/valtimo/config/assets/core/nl.json]:
```suggestion
      "description": "Er zijn geen logregels die overeenkomen met je zoekcriteria",
```

File: projects/valtimo/logging/src/lib/components/log-search/log-search.component.scss [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+.valtimo-log-search {
+  flex-direction: column;
+  padding: 16px;
+  max-width: 880px;
+
+  &,
```
Comment (mbritense) [inline projects/valtimo/logging/src/lib/components/log-search/log-search.component.scss:22]: is this on purpose?

File: projects/valtimo/logging/src/lib/components/log-search/log-search.component.scss [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    align-items: flex-end;
+  }
+
+  &__label {
+    font-size: var(--cds-label-01-font-size);
+    color: var(--cds-text-secondary);
+  }
+}
+
+::ng-deep .cds--date-picker.cds--date-picker--single .cds--date-picker__input {
```
Comment (mbritense) [inline projects/valtimo/logging/src/lib/components/log-search/log-search.component.scss]: :host ::ng-deep is a bit safer imo

File: projects/valtimo/logging/src/lib/components/log-search/log-search.component.ts [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    afterTimestamp: this.fb.control<string>(''),
+    properties: this.fb.array([
+      this.fb.group({
+        key: this.fb.control<string>(''),
+        value: this.fb.control<string>(''),
+      }),
+    ]),
+  });
+
+  public logLevelItems: ListItem[] = [
```
Comment (mbritense) [inline projects/valtimo/logging/src/lib/components/log-search/log-search.component.ts]: better make this Partial<ListItem> so you don't have to do "selected: false" for each one, you're mapping this later anyway

File: projects/valtimo/logging/src/lib/components/log-search/log-search.component.scss (hunk 2 above)
Comment (teodor-ritense) [reply projects/valtimo/logging/src/lib/components/log-search/log-search.component.scss:22]: Yes, it is, there's common styling between the parent and the children

**Review by mbritense — APPROVED** [review]

**Pull request merged**